### PR TITLE
arch: arm: Refactor CONFIG_CORTEX_M

### PR DESCRIPTION
**Pull request by galak**
Clean up Kconfig so each SoC just selects the specific Cortex-M
implementaiton rather than having to select both CORTEX_M and
CORTEX_{M0, M3, M4, etc.}.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: arch/arm/soc/ti_simplelink/cc32xx/Kconfig.series [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 config  SOC_SERIES_CC32XX
 	bool "TI SimpleLink Family"
-	select CPU_CORTEX_M
```
Comment (GAnthony) [inline arch/arm/soc/ti_simplelink/cc32xx/Kconfig.series]: LGTM, thanks!

**Review by GAnthony — DISMISSED** [review]

File: arch/arm/core/Kconfig [hunk 2]
```diff
@@ -69,9 +69,7 @@ endchoice
 
 endmenu
 
-if CPU_CORTEX_M
```
Comment (MaureenHelm) [inline arch/arm/core/Kconfig]: Won't this expose configs like `CPU_CORTEX_M_HAS_BASEPRI` and `CPU_CORTEX_M_HAS_VTOR` to non-Cortex-M arches? Should we move them into the "ARM Cortex-M options" menu?

Comment (ioannisg) [reply arch/arm/core/Kconfig]: Or, perhaps keep the conditional here, as follows:

`if CPU_CORTEX_M0 || CPU_CORTEX_M0PLUS || CPU_CORTEX_M3 || CPU_CORTEX_M4 || CPU_CORTEX_M23 || CPU_CORTEX_M33 || CPU_CORTEX_M7`

I know, this is a bit ugly, but it is not likely that we have to touch this line too often.

I think that we need to keep the conditional "source", for cortex_m/Kconfg; we should try not to expose any of the K-options defined there, if we don't have a Cortex-M CPU.

**Review by ioannisg — DISMISSED** [review]
Please, comment on my proposal to re-work the CPU_CORTEX_Mxx defines

**Comment by galak** [comment]
> Please, comment on my proposal to re-work the CPU_CORTEX_Mxx defines

Take a look at my update, I want to avoid patterns like depends on M0 || M0+ || M3 || M4 || M23 || M33 || M7, so want to keep the CPU_CORTEX_M option around.  But we can isolate the bulk of things

**Comment by ioannisg** [comment]
@galak It looks OK to me. I think we need to stress (e.g. with a comment) in ./arm/core/cortex_m/Kconfig, that any definitions shall be placed after `#if CPU_CORTEX_M`.

**Comment by galak** [comment]
> @galak It looks OK to me. I think we need to stress (e.g. with a comment) in ./arm/core/cortex_m/Kconfig, that any definitions shall be placed after #if CPU_CORTEX_M.

Ok, I'll add a comment about that.

**Comment by galak** [comment]
@ioannisg added comment block, see if that looks good.

**Comment by ioannisg** [comment]
Looks good.

File: arch/arm/core/cortex_m/Kconfig [hunk 3]
```diff
@@ -259,3 +271,5 @@ config SW_VECTOR_RELAY
 	  (or an Armv8-M baseline core) with no VTOR and no other hardware
 	  relocation table mechanisms.
 endmenu
+
+endif # CORTEX_M
```
Comment (ioannisg) [inline arch/arm/core/cortex_m/Kconfig]: @galak minor: CPU_CORTEX_M

**Review by ioannisg — DISMISSED** [review]

File: arch/arm/core/cortex_m/Kconfig (hunk 3 above)
Comment (galak) [reply arch/arm/core/cortex_m/Kconfig]: Will Fix

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
@galak This PR has accidentally removed the `select ARMV7_M_ARMV8_M_FP if CPU_HAS_FPU`
configurations in CM4, CM7, and CM33, originaly merged to "arm" by #6390 . Could you revert & fix this one, or shall we make a new PR?

**Pull request merged**